### ISSUE TITLE
Add in-browser demo on documentation

### ISSUE THREAD

**Issue by bollwyvl** [ipython/ipython#13936]
## References

- from #13828

## Elevator Pitch

Offer an IPython experience in the browser when viewing the documentation website.

## Motivation

This gives prospective users a way to _try_ instead of _watch_ or _read_ about IPython, while using fewer community resources such as mybinder.

PR authors and reviewers would also be able to review changes in a very limited environment, e.g. in a custom python, inside emscripten, inside a webworker, inside... whatever a browser happens to be. 

## Design Ideas

JupyterLite (well, specifically `pyolite`) has intentionally made it possible to overload core dependencies such as `ipython` _without_ a complex `nodejs`-based toolchain, while being able to be extended by other `pip`-installable dependencies.

As such, a one-day hack, and a few more docs dependencies, would get to a working JupyterLab-derived console, based on [JupyterLite's `repl` app](https://jupyterlite.readthedocs.io/en/latest/quickstart/embed-repl.html), but running the as-built-for-the-PR python package. 

From a cold cache, getting to interactive pixels clocks in at about 14mb downloaded by the client (the size of a decently-sized youtube video) up to `ipython` + `ipykernel`, and would add a few seconds to the docs build. Subsequent loads can be cached with an extra-resilient ServiceWorker, which also offers the ability to preload content.

## Future Work

Getting a real xterm.js-based [terminal app](https://github.com/jupyterlite/jupyterlite/issues/717), which filled the whole  screen (but probably `iframe`) would be a bigger lift, and i'd probably only do it _if_ there was interest from here. A first cut would be akin to a js-based [jupyter_console](https://github.com/jupyter/jupyter_console), using `ipykernel`, as this would work for any kernel.

Going all the way, without `ipykernel`, would require more-or-less directly implementing against `pyodide`, and wouldn't use any jupyter kernel messaging... though it probably _would_ reuse the in-browser jupyter contents, etc. to lazily pre-populate the filesystem. But this is probably blocked by https://github.com/pyodide/pyodide/issues/2750.

**Comment by bollwyvl** [comment]
With an eye towards this feeling snappy in the browser: every byte on the wire and entry in `sys.modules` counts to a greater degree in the browser than in a "real" computer, even with caching at various levels of robustness.

Some measures:

- 0.2mb: move all of the tests and the nose plugin out of the wheel
  - but keep them in the sdist 
- lazier, less side-effectful imports of 
  - 1.8mb of `jedi` + `parso` 
  - 0.5mb of `sqlite3`
  - 0.4mb of `prompt_toolkit`

Some of this is being abused/explored over on [jupyterlite](https://github.com/jupyterlite/jupyterlite/pull/979) where we've gotten the "novel" parts of starting a kernel to ~9.5mb 

This would of course _also_ improve cold boot time for real tty/kernel sessions (and to a lesser extent, install time, depending on your bandwidth), which is nice.
